packages/fuels/tests/harness.rs

```diff
@@ -3716,3 +3720,88 @@ async fn test_rust_result_can_be_encoded() -> Result<(), Box<dyn std::error::Err
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_script_interface() -> Result<(), Error> {
+    let wallet = launch_provider_and_get_wallet().await;
+
+    let contract_id = Contract::deploy(
+        "../../packages/fuels/tests/test_projects/contract_test/out/debug/contract_test.bin",
+        &wallet,
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_coins = wallet
+        .get_provider()?
+        .get_contract_balances(&contract_id)
+        .await?;
+    assert!(contract_coins.is_empty());
+
+    let amount = 100;
+    let asset_id = Default::default();
+    let tx_parameters = TxParameters::default();
+    let zeroes = Bytes32::zeroed();
+    let plain_contract_id: ContractId = contract_id.clone().into();
+
+    let mut inputs = vec![Input::contract(
+        UtxoId::new(zeroes, 0),
+        zeroes,
+        zeroes,
+        TxPointer::default(),
+        plain_contract_id,
+    )];
+    inputs.extend(
+        wallet
+            .get_asset_inputs_for_amount(asset_id, amount, 0)
+            .await?,
+    );
+
+    let outputs = vec![
+        Output::contract(0, zeroes, zeroes),
+        Output::change(wallet.address().into(), 0, asset_id),
+    ];
+
+    let script_data: Vec<u8> = [
+        plain_contract_id.to_vec(),
+        amount.to_be_bytes().to_vec(),
+        asset_id.to_vec(),
+    ]
+    .into_iter()
+    .flatten()
+    .collect();
+
+    let script = vec![
+        Opcode::gtf(0x10, 0x00, GTFArgs::ScriptData),
+        Opcode::ADDI(0x11, 0x10, ContractId::LEN as u16),
+        Opcode::LW(0x11, 0x11, 0),
+        Opcode::ADDI(0x12, 0x11, WORD_SIZE as u16),
+        Opcode::TR(0x10, 0x11, 0x12),
+        Opcode::RET(REG_ONE),
+    ]
+    .into_iter()
+    .collect();
+
+    ScriptBuilder::new()
+        .set_gas_price(tx_parameters.gas_price)
+        .set_gas_limit(tx_parameters.gas_limit)
+        .set_maturity(tx_parameters.maturity)
+        .set_script(script)
+        .set_script_data(script_data)
+        .set_inputs(inputs.to_vec())
+        .set_outputs(outputs.to_vec())
+        .set_amount(amount)
+        .build(&wallet)
+        .await?
+        .call(wallet.get_provider()?)
+        .await?;
```

Comment:
I think this test is not "hard" enough: for instance I would like to see that providing a `gas_price` or `gas_limit` too low makes it fail. But maybe I'm being paranoid since they are essentially wrappers around other, well-tested functions.

Reply:
This is a test to test if everything works as it should. I have already tested the situations you mention. They can be added if there is a demand.